src/plugins/utils/http-utils.ts

```diff
@@ -36,7 +36,17 @@ export type HttpPluginConfig = {
     // Applications that wish to link their client and server traces by enabling
     // the X-Amzn-Trace-Id header should test their applications before enabling
     // it in a production environment.
-    addXRayTraceIdHeader: boolean;
+    addXRayTraceIdHeader: boolean | RegExp[];
+};
+
+export const shouldAddXrayTraceIdHeader = (
```

Comment:
suggestion: Rename `isXRayTraceIdHeaderEnabled` or `isTraceIdHeaderEnabled`.

Reply:
suggestion: Remove some redundancy between the XhrPlugin tests and FetchPlugin tests by unit testing this function.